src/Redistribution.sol

```diff
@@ -679,6 +674,20 @@ contract Redistribution is AccessControl, Pausable {
 
     // ----------------------------- Reveal ------------------------------
 
+    /**
+     * @notice Helper function to get this node reveal in commits
+     * @dev
+     */
+    function findCommit(bytes32 _overlay, bytes32 _commitHash, uint256 _length) internal view returns (uint256 id) {
+        id = type(uint256).max;
+        for (uint256 i = 0; i < _length; i++) {
+            if (currentCommits[i].overlay == _overlay && _commitHash == currentCommits[i].obfuscatedHash) {
+                id = i;
+            }
+        }
+        return id;
```

Comment:
throw the exception here.

Reply:
yes, all of these good ideas, much better code, and less gas consumption as well. thnx